Undefined symbols for architecture x86_64

I am using the cocoapods objective c version in my project and linking gives me hundreds of errors. Such as:

Undefined symbols for architecture x86_64:
"realm::BpTreeBase::write_subtree(realm::Array const&, unsigned long, unsigned long, unsigned long, realm::BpTreeBase::SliceHandler&, realm::_impl::OutputStream&)", referenced from:
      realm::BpTree<long long, false>::write(unsigned long, unsigned long, unsigned long, realm::_impl::OutputStream&) const in libRealm.a(RLMResults.o)
      realm::BpTree<long long, false>::write(unsigned long, unsigned long, unsigned long, realm::_impl::OutputStream&) const in librealm.a(RLMResults.o)
      realm::BpTree<long long, false>::write(unsigned long, unsigned long, unsigned long, realm::_impl::OutputStream&) const in libRealm.a(object_store.o)

osx 10.10, every other pod seems to work fine though. Any idea what linker option is missing? Thanks!

--- a/Realm.podspec
+++ b/Realm.podspec
@@ -46,7 +46,7 @@ Pod::Spec.new do |s|
   s.ios.vendored_library    = 'core/librealm-ios.a'
 
   s.osx.deployment_target   = '10.9'
-  s.osx.vendored_library    = 'core/librealm.a'
+  s.osx.vendored_library    = 'core/librealm-osx.a'
 
   if s.respond_to?(:watchos)
     s.watchos.deployment_target = '2.0'

--- a/build.sh
+++ b/build.sh
@@ -644,6 +644,7 @@ case "$COMMAND" in
     "cocoapods-setup")
         if [[ "$2" != "without-core" ]]; then
             sh build.sh download-core
+            mv core/librealm.a core/librealm-osx.a
         fi
 
         # CocoaPods won't automatically preserve files referenced via symlinks

